spiped/dispatch.c

```diff
@@ -28,6 +28,7 @@ struct accept_state {
 	double timeo;
 	void * accept_cookie;
 	void * dnstimer_cookie;
+	void * conn_cookie;
```

Comment:
spiped might have lots of connections.  Why are you holding a pointer to just one?